robot/robot.go

```diff
@@ -57,6 +58,9 @@ func (r *Robot) Move(steps int) error {
 	if r.EnvMap.Map == "" {
 		return fmt.Errorf("The robot can't move because no environmental map is given")
 	}
+	if !r.canMoveToPosition(steps) {
+		return fmt.Errorf("The robot can't move %v steps because their are obstacles in the way.Here's the current map: %v", steps, mapWithRobot(r))
+	}
 	newPosition := r.Position
 	switch r.Position.Orientation {
```

Comment:
Can you think of a way to remove code duplication of the following lines and the new canMoveToPosition(int) method?